File: 2016/07/two.py
lines = open("input.txt").read().split("\n")
m = 0
for line in lines:
    llc = " "
    lc = " "
    sb = False
    inside = set()
    outside = set()
    for c in line:
        if c == "[":
            sb = True
        elif c == "]":
            sb = False
        elif c == llc and c != lc:
            if sb: inside.add(lc + c)
            else: outside.add(c + lc)
        llc = lc
        lc = c
    for i in inside:
        if i in outside:
            m += 1
            break
print(m)

File: 2016/02/two.go
package main

import (
    "fmt"
    "os"
    "strings"
)

func main() {
    x, y := 2, 2

    data, err := os.ReadFile("input.txt")
    if err != nil { panic(err) }

    input := string(data)
    lines := strings.Split(strings.Trim(input, "\n"), "\n")

	code := ""
    keypad := [5]string{"  1  ", " 234 ", "56789", " ABC ", "  D  "}

    for _, line := range lines {
		for _, char := range line {
			if char == 'U' && y > max(x - 2, 2 - x) { y-- }
			if char == 'D' && y < 4 - max(x - 2, 2 - x) { y++ }
			if char == 'L' && x > max(y - 2, 2 - y) { x-- }
			if char == 'R' && x < 4 - max(y - 2, 2 - y) { x++ }
		}

		code += string(keypad[y][x])
    }
    
    fmt.Println(code)
}